logsmanagement/unit_test/unit_test.c

```diff
@@ -0,0 +1,171 @@
+/** @file unit_test.h
+ *  @brief Includes unit tests for the logs management project
+ *
+ *  @author Dimitris Pantazis
+ */
+
+#include "unit_test.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include "../circular_buffer.h"
+#include "../compression.h"
+#include "../parser.h"
+#include "../query.h"
+
+#define SEVERAL_LOG_RECORDS "\
+127.0.0.1 - - [30/Jun/2022:16:43:51 +0300] \"GET / HTTP/1.0\" 200 11192 \"-\" \"ApacheBench/2.3\"\n\
+192.168.2.1 - - [30/Jun/2022:16:43:51 +0300] \"PUT / HTTP/1.0\" 400 11192 \"-\" \"ApacheBench/2.3\"\n"
+
+static int test_compression_decompression() {
+    int errors = 0;
+    fprintf(stderr, "%s():\n", __FUNCTION__);
+
+    Circ_buff_item_t item;
+    item.text_size = sizeof(SEVERAL_LOG_RECORDS);
+    fprintf(stderr, "Testing LZ4_compressBound()...\n");
+    size_t required_compressed_space  = LZ4_compressBound(item.text_size);
+    if(!required_compressed_space){
+        fprintf(stderr, "- Error while using LZ4_compressBound()\n");
+        return ++errors;
+    }
+
+    item.data_max_size = item.text_size + required_compressed_space;
+    item.data = mallocz(item.data_max_size);
```

Comment:
please, use `callocz` instead `mallocz` or set a `NULL` character in `item.data`.

Reply:
I don't think it's needed as we know exactly the size of the `item.data` data every time we operate on it.